src/Microsoft.Restier.Core/Api.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Restier.Core
+{
+    public sealed class Api<TKey> where TKey : class
+    {
```

Comment:
This class could be eliminated entirely, and let ApiConfiguration manage configuration cache.
Or maybe we'd leave this and move ApiBuilder cache here?